lib/publiccloud/instance.pm

```diff
@@ -264,6 +264,33 @@ sub upload_log {
     assert_script_run("test -d '$tmpdir' && rm -rf '$tmpdir'");
 }
 
+=head2 verify_upload_logs
+
+    verify_upload_logs();
+
+Check files status and upload logs ok to UI.
+=cut
+
+sub verify_upload_logs {
+    my ($self, $dir, @files) = @_;
+    my @logs = ();
+    my @missing_logs = ();
+
+    # Select existing files
+    foreach (@files) {
+        my $res = $self->ssh_script_run(cmd => "test -e $dir/" . $_, quiet => 1, proceed_on_failure => 1);
+        $res = isok($res) ? push(@logs, $_) : push(@missing_logs, $_);
+    }
+    record_info("Logs missing", "Files not found in $dir: " . join(', ', @missing_logs)) if ($#missing_logs > 0);
```

Comment:
Do we really need to know which logs are missing?  We can infer this information from the `Logs & Assets` tab.

By running `$self->ssh_script_run(cmd => "ls @logs", ...);` we can get a list of existing logs.

Reply:
It is a no-cost more information IMHO useful to have soon, instead of infer.

Reply:
moreover, we're assigning the file list in input and we expect all are ok, so also I expect that ecxeptions will be notified, not hidden, that also caused this poo's issue not be noted until now.

Reply:
There are NO more ssh than needed: I check existing file via ssh, then I relize that one is present or not, then I store those ok, but  those `not` too **at no-or-low cost**.

Reply:
You can easily obtain the list of existing files with a simple `$self->ssh_script_run(cmd => "ls @logs", proceed_on_failure => 1);`

Reply:
In your VRs I can see that `prepare_instance-cloudregister.log.txt` is uploaded but yet it is reported as missing.

Reply:
And on the top of that, the file is uploaded but yet empty. So the existance of a file is nice, but uploading empty files is not useful

Reply:
After those VRs you mention and I also noted, I changed the code, last one is ok. 
About empty log I can change the test option.

Reply:
The discussed code status was updates with not_empty and tar format, please check last status.